site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/AutoGroupsController.java

```diff
@@ -296,6 +298,25 @@ public String showStep4(Model model, @ModelAttribute AutoGroupsForm autoGroupsFo
                     }
                 }
             }
+
+            if (useManuallyAddedUsers) {
+                List<Member> allSiteMembers = site.getMembers().stream().filter(m -> selectedRoleList.contains(m.getRole().getId())).collect(Collectors.toList());
+                // Build the section list.
+                List<Group> sectionList = site.getGroups().stream()
+                    .filter(
+                        group -> group.getProperties().getProperty(Group.GROUP_PROP_WSETUP_CREATED) == null || 
+                        !Boolean.valueOf(group.getProperties().getProperty(Group.GROUP_PROP_WSETUP_CREATED)).booleanValue()
+                    ).collect(Collectors.toList());
+                List<Member> siteSectionMembers = new ArrayList<Member>(); 
+                sectionList.forEach(section -> {siteSectionMembers.addAll(section.getMembers());});
+                for (Member member : allSiteMembers) {
+                    if (siteSectionMembers.stream().noneMatch(sectionMember -> sectionMember.getUserId().equals(member.getUserId()))) {
+                        filteredMembers.add(member);
+                    }
+                }
+                
+            }
```

Comment:
I think I've written code in the past that gets the non-provided users in a site, which IIRC is less complex than this. I'll try and find it and post back here.